### ISSUE TITLE
can i analysis a file without type checking? 

### ISSUE THREAD

**Issue by wzl-1997** [MikePopoloski/slang#1156]
hello,  i'm developing a language server based on slang. 
I want to use slang to analysis a single file with out type check, such as error: `use of undeclared identifier 'xxx'`. 
I just want to diagnose syntax error, such as `expected ';' `
I has tried the --lint-only option, and it seems not work. 
Dose slang support this feature? If so, how should i do, pass any options to slang? 
Thanks

**Comment by likeamahoney** [comment]
Hello, you should use `--parse-only` cli options I suppose.

**Comment by MikePopoloski** [comment]
Yes, `--parse-only` will perform basic syntax checking. That does not include name resolution though; you need to do full type checking to also get name resolution errors reported.
